lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -895,7 +895,7 @@ def exec_module(self, **kwargs):
                     vm_dict['properties']['storageProfile']['osDisk']['caching'] = self.os_disk_caching
 
                 if self.os_disk_size_gb and \
-                   self.os_disk_size_gb != vm_dict['properties']['storageProfile']['osDisk']['diskSizeGB']:
+                   self.os_disk_size_gb != vm_dict['properties']['storageProfile']['osDisk'].get('diskSizeGB'):
```

Comment:
What's the situation where this field is not present? Is it managed disk with VM default size? I'm wondering if we should treat the "missing" case differently or not...